src/custom/state/orders/updater.ts

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { useActiveWeb3React } from 'hooks'
+import { useAddPopup, useBlockNumber } from 'state/application/hooks'
+import { AppDispatch } from 'state'
+// import { removeOrder, /* fulfillOrder, */ OrderFromApi } from './actions'
+import { utils } from 'ethers'
+import { Web3Provider } from '@ethersproject/providers'
+import { Log, Filter } from '@ethersproject/abstract-provider'
+import { useLastCheckedBlock } from './hooks'
+import { updateLastCheckedBlock } from './actions'
+// import { PartialOrdersMap } from './reducer'
+
+// example of event watching + decoding without contract
+const transferEventAbi = 'event Transfer(address indexed from, address indexed to, uint amount)'
+const ERC20Interface = new utils.Interface([transferEventAbi])
+
+const TransferEvent = ERC20Interface.getEvent('Transfer')
+
+const TransferEventTopics = ERC20Interface.encodeFilterTopics(TransferEvent, [])
+
+// const decodeTransferEvent = (transferEventLog: Log) => {
+//   return ERC20Interface.decodeEventLog(TransferEvent, transferEventLog.data, transferEventLog.topics)
+// }
+
+type RetryFilter = Filter & { fromBlock: number; toBlock: number }
+
+const constructGetLogsRetry = (provider: Web3Provider) => {
+  // tries fetching logs
+  // if breaks, retries on half the range
+  const getLogsRetry = async ({ fromBlock, toBlock, ...rest }: RetryFilter): Promise<Log[]> => {
+    try {
+      console.log(`EventUpdater::Getting logs fromBlock: ${fromBlock} toBlock ${toBlock}`)
+      const logs = await provider.getLogs({
+        fromBlock,
+        toBlock,
+        ...rest
+      })
+      // console.log('logs', logs)
+      return logs
+    } catch (error) {
+      console.error(`Error getting logs fromBlock: ${fromBlock} toBlock ${toBlock}`, error)
+
+      // expect -- RPC Error: query returned more than 10000 results --
+      // if a different error - rethrow
+      if (!error?.message?.includes('query returned more than')) throw error
+
+      const midBlock = Math.floor((toBlock + fromBlock) / 2)
+
+      const [beforeMidLogs, afterMidLogs] = await Promise.all([
```

Comment:
nice feature, but what if it fails for ratelimits?

This could in a worst scenarion split in one RPC call per block? (half of the half of half..... ---> 1 block)

Reply:
If it fails for any reason other than `query returned more than` (L46) then we rethrow and the whole thing breaks, that is to say nothing happens, error in console, fetch will resume on next block

Reply:
Or user grows impatient, refreshes, fetches anew